src/redis.cr

```diff
@@ -246,8 +272,8 @@ class Redis
     close
     raise ex
   ensure
-    if _client = @client
-      _client.strategy = Redis::Strategy::SingleStatement.new(_client.connection)
+    if @connection
```

Comment:
`if _connection = @connection` more correct here, because @connection potentially, can become nil(in multithread env) and `@connection.not_nil!` would raise.

Reply:
During each command (including `multi` and `pipelined`), the `Redis` object must be owned/used by a single thread only. Therefore I see no multithread issue here.

This rule is followed by `Redis::PooledClient`.